src/bin/ion/commands/beta/to.rs

```diff
@@ -0,0 +1,185 @@
+        io::copy(&mut io::stdin(), &mut writer)
+            .with_context(|| "Failed to copy STDIN to a temp file.")?;
+        // Get our file handle back from the BufWriter
+        input_file = writer
+            .into_inner()
+            .with_context(|| "Failed to read from temp file containing STDIN data.")?;
+        convert(&mut input_file, &mut output, format)?;
+    }
+
+    output.flush()?;
+    Ok(())
+}
+
+pub fn convert(file: &mut File, output: &mut Box<dyn Write>, format: &str) -> Result<()> {
+    // NOTE: mmap logic is copied from inspect.inspect_file().
+
+    // mmap involves operating system interactions that inherently place its usage outside of Rust's
+    // safety guarantees. If the file is unexpectedly truncated while it's being read, for example,
+    // problems could arise.
+    let mmap = unsafe {
+        MmapOptions::new()
+            .map(&file)
+            .with_context(|| "Could not mmap ")?
+    };
+
+    // Treat the mmap as a byte array.
+    let ion_data: &[u8] = &mmap[..];
+    let mut iter = element_reader()
+        .iterate_over(ion_data)
+        .expect("Cannot iterate.");
+    match format {
+        "json" => {
+            while let Some(x) = iter.next() {
+                let element = x.expect("Failed to parse.");
+                write!(output, "{}\n", to_json_value(element).to_string())
+                    .expect("Failed to write JSON");
+            }
+        }
+        _ => {
+            unimplemented!("Unsupported format.")
+        }
+    };
+    Ok(())
+}
+
+fn to_json_value(element: Element) -> JsonValue {
+    if element.is_null() {
+        JsonValue::Null
+    } else {
+        match element.ion_type() {
+            IonType::Null => JsonValue::Null,
+            IonType::Boolean => JsonValue::Bool(element.as_bool().unwrap()),
+            IonType::Integer => JsonValue::Number(
+                Number::from_str(&*element.as_integer().unwrap().to_string()).unwrap(),
+            ),
+            IonType::Float => {
+                JsonValue::Number(Number::from_f64(element.as_f64().unwrap()).unwrap())
+            }
+            IonType::Decimal => JsonValue::Number(
+                Number::from_str(
+                    element
+                        .as_decimal()
+                        .unwrap()
+                        .to_string()
+                        .replace("d", "e")
+                        .as_str(),
+                )
+                .unwrap(),
+            ),
+            IonType::Timestamp => JsonValue::String(element.as_timestamp().unwrap().to_string()),
+            IonType::Symbol | IonType::String => {
+                JsonValue::String(element.as_str().unwrap().into())
+            }
+            IonType::Clob => {
+                JsonValue::String(element.to_string().replace("{{\"", "").replace("\"}}", ""))
+            }
+            IonType::Blob => {
+                JsonValue::String(element.to_string().replace("{{", "").replace("}}", ""))
```

Comment:
That's the technique I originally used, but due to the fact that clobs have [special rules](https://amazon-ion.github.io/ion-docs/guides/cookbook.html#down-converting-to-json) for JSON down-conversion, I ended up needing to copy logic from [raw_text_writer](https://github.com/amazon-ion/ion-rust/blob/main/src/text/raw_text_writer.rs#L612). If that were exposed as a function we could reuse it, but we probably don't want to expose that. Ultimately since we [only](https://github.com/amazon-ion/ion-rust/blob/main/src/text/raw_text_writer.rs#L633) write clobs using the short-string syntax right now, I went with this for both clobs and blobs for consistency.

Reply:
#45